File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int mini=nums[0];
        int maxDiff=-1;
        for(int i=1;i<nums.length;i++) {
            if(nums[i]>mini){
                maxDiff=Math.max(maxDiff,nums[i]-mini);
            }else{
                mini=nums[i];
            }
        }
        return maxDiff;
    }
}

File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.py
class Solution(object):
    def maximumDifference(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mini=nums[0]
        maxdiff=-1
        for i in range(1,len(nums)):
            if nums[i]>mini:
                maxdiff=max(maxdiff,nums[i]-mini)
            else:
                mini=nums[i]
        return maxdiff
